Modules/posixmodule.c

```diff
@@ -5553,6 +5520,104 @@ os__path_normpath_impl(PyObject *module, path_t *path)
     return result;
 }
 
+#ifdef MS_WINDOWS
+#define IS_MS_WINDOWS 1
+#else
+#define IS_MS_WINDOWS 0
+#endif
```

Comment:
Is there a better way to do this?

Reply:
I'd prefer for Windows to simply use `nonstrict`. Fall back to a manual implementation if the path contains null characters or if `_PyOS_getfullpathname()` fails. The strict-mode path-length check is fundamentally flawed anyway. Ultimately, success depends on whether the final resolved length of the full path can be stored as a `UNICODE_STRING`, which can have no more than 32766 characters (i.e. `UNICODE_STRING_MAX_CHARS - 1`). That in turn depends on the length of the working directory, given the input path is relative.

I'd prefer to fall back on a manual implementation in C, but it's a bit more complicated than the POSIX implementation. It needs to check for a simple relative path, a drive-relative path, or a rooted relatived path. The drive-relative case gets the working directory to use from `_PyOS_getfullpathname()` for the drive. Otherwise use `posix_getcwd()`. A simple relative path just gets concatenated to the working directory. For a drive-relative path, the rest after the drive gets concatenated to the working directory on the drive. A rooted path gets concatenated to just the drive of the working directory, which requires `_Py_skiproot()` in general to handle UNC paths.

Reply:
OK, I'll check it manually in Python then, implementing it in C seems a bit too complicated right now.